### DISCUSSION TITLE
[Feature Request] AddSourceIf and SubSourceIf

### DISCUSSION THREAD

**Discussion by Blazekickn** [RetroAchievements/RAIntegration#897] [Ideas]
AndNext is a great feature, but one of my biggest problems with it is that it can't be used in the middle of an AddSource/SubSource chain to enable/disable certain addresses without needing alt groups. AddSourceIf, SubSourceIf would help fill that role without changing the original AndNext logic. I recently had an achievement concept that is impossible to code properly without this feature, as the amount of alt groups required would be way too much for the character limit.

**Comment by SporyTike** [comment]
An actual example could be https://retroachievements.org/game/10617

This game has a championship mode which is several races in a row and you get points each race, every race result (finish time and point reward) is stored in memory. However, unlike many other racing games you can get disqualified in a race due to certain actions and the disqualification penalty is equal to +10 minutes on the racing timer, however the game writes the time when the player was disqualified into memory instead of the penalty itself, since that is tracked using the individual race point results. The last place always gets 1 point, but if you get disqualified you get 0 points for that race, so if the player got 0 points in a race, it is shown as disqualified in the championship results.

If the player gets enough points in the first few races it is possible to get disqualified in the last few races and still win, but due to the current limitations the leaderboards needs to force that all races are successfully finished to avoid a race  adding an impossible time (e.g. just a few seconds) because the player got disqualified early.

The solution for that would be to have some code like this for each race time value:
`SourceIf  Mem   RacePoints =  Value 0x00`
`AddSource Value '10 Minutes'`
`SourceIf  Mem   RacePoints != Value 0x00`
`AddSource Mem   RaceTime`
